components/formats-api/src/loci/formats/SubResolutionFormatReader.java

```diff
@@ -0,0 +1,1891 @@
+  @Override
+  public void setMetadataStore(MetadataStore store) {
+    FormatTools.assertId(currentId, false, 1);
+    if (store == null) {
+      throw new IllegalArgumentException("Metadata object cannot be null; " +
+        "use loci.formats.meta.DummyMetadata instead");
+    }
+    metadataStore = store;
+  }
+
+  /* @see IFormatReader#getMetadataStore() */
+  @Override
+  public MetadataStore getMetadataStore() {
+    return metadataStore;
+  }
+
+  /* @see IFormatReader#getMetadataStoreRoot() */
+  @Override
+  public Object getMetadataStoreRoot() {
+    FormatTools.assertId(currentId, true, 1);
+    return getMetadataStore().getRoot();
+  }
+
+  /* @see IFormatReader#getUnderlyingReaders() */
+  @Override
+  public IFormatReader[] getUnderlyingReaders() {
+    return null;
+  }
+
+  /* @see IFormatReader#isSingleFile(String) */
+  @Override
+  public boolean isSingleFile(String id) throws FormatException, IOException {
+    return true;
+  }
+
+  /* @see IFormatReader#getRequiredDirectories(String[]) */
+  @Override
+  public int getRequiredDirectories(String[] files)
+    throws FormatException, IOException
+  {
+    return 0;
+  }
+
+  /* @see IFormatReader#getDatasetStructureDescription() */
+  @Override
+  public String getDatasetStructureDescription() {
+    return datasetDescription;
+  }
+
+  /* @see IFormatReader#hasCompanionFiles() */
+  @Override
+  public boolean hasCompanionFiles() {
+    return hasCompanionFiles;
+  }
+
+  /* @see IFormatReader#getPossibleDomains(String) */
+  @Override
+  public String[] getPossibleDomains(String id)
+    throws FormatException, IOException
+  {
+    return domains;
+  }
+
+  /* @see IFormatReader#getDomains() */
+  @Override
+  public String[] getDomains() {
+    FormatTools.assertId(currentId, true, 1);
+    return domains;
+  }
+
+  /* @see IFormatReader#getOptimalTileWidth() */
+  @Override
+  public int getOptimalTileWidth() {
+    FormatTools.assertId(currentId, true, 1);
+    return getSizeX();
+  }
+
+  /* @see IFormatReader#getOptimalTileHeight() */
+  @Override
+  public int getOptimalTileHeight() {
+    FormatTools.assertId(currentId, true, 1);
+     int bpp = FormatTools.getBytesPerPixel(getPixelType());
+     int maxHeight = (1024 * 1024) / (getSizeX() * getRGBChannelCount() * bpp);
+     return (int) Math.min(maxHeight, getSizeY());
+  }
+
+  // -- Sub-resolution API methods --
+
+  @Override
+  public int seriesToCoreIndex(int s) {
+    if (hasFlattenedResolutions()) {
+      return s;
+    } else {
+      return core.flattenedIndex(s, 0);
+    }
+  }
+
+  @Override
+  public int coreIndexToSeries(int index) {
+    if (hasFlattenedResolutions()) {
+      return index;
+    } else {
+      return core.flattenedIndexes(index)[0];
+    }
+  }
+
+  /* @see IFormatReader#getResolutionCount() */
+  @Override
+  public int getResolutionCount() {
+    FormatTools.assertId(currentId, true, 1);
+
+    if (hasFlattenedResolutions()) {
+      return 1;
+    }
```

Comment:
close up linebreak to `} else {`

Reply:
While I'd like this, it's actually the other way around in terms of code style!